test/unit/util/sfdcUrlTest.ts

```diff
@@ -0,0 +1,226 @@
+      const url = new SfdcUrl('https://www.ghostbusters.com');
+      expect(url.isSalesforceDomain()).to.be.false;
+    });
+
+    it('is allowlist host', () => {
+      const url = new SfdcUrl('https://developer.salesforce.com');
+      expect(url.isSalesforceDomain()).to.be.true;
+    });
+
+    it('falsy', () => {
+      try {
+        const url = new SfdcUrl(undefined);
+        assert(url, 'should throw');
+        expect(url.isSalesforceDomain()).to.be.false;
+      } catch (e) {
+        expect(e.name).to.equal('TypeError');
+      }
+    });
+  });
+
+  describe('internal domains', () => {
+    it('stm is internal but not local', () => {
+      const url = new SfdcUrl('https://inttestdevhub02-dev-ed.lightning.stmfa.stm.force.com/');
+      expect(url.isInternalUrl()).to.equal(true);
+      expect(url.isLocalUrl()).to.equal(false);
+    });
+    it('pc.rnd is internal but not local', () => {
+      const url = new SfdcUrl('https://alm-cidevhubora3test1core4.test1.lightning.pc-rnd.force.com/');
+      expect(url.isInternalUrl()).to.equal(true);
+      expect(url.isLocalUrl()).to.equal(false);
+      expect(url.isInternalUrl()).to.equal(true);
+      expect(url.isLocalUrl()).to.equal(false);
+    });
+    it('localhost domain is both internal and local, and tolerates local host ports', () => {
+      const url = new SfdcUrl('https://scorpio-ryan-2873-dev-ed.my.localhost.sfdcdev.salesforce.com:6109');
+      expect(url.isInternalUrl()).to.equal(true);
+      expect(url.isLocalUrl()).to.equal(true);
+    });
+  });
+
+  describe('checkLightningDomain', () => {
+    beforeEach(() => {
+      $$.SANDBOX.stub(MyDomainResolver.prototype, 'resolve').resolves(TEST_IP);
+    });
+
+    afterEach(() => {
+      $$.SANDBOX.restore();
+    });
+
+    it('return true for internal urls', async () => {
+      const url = new SfdcUrl('https://my-domain.stm.salesforce.com');
+      const response = await url.checkLightningDomain();
+      expect(response).to.be.true;
+    });
+
+    it('return true for urls that dns can resolve', async () => {
+      const url = new SfdcUrl('https://login.salesforce.com');
+      const respose = await url.checkLightningDomain();
+      expect(respose).to.be.true;
+    });
+
+    it('throws on domain resolution failure', async () => {
+      $$.SANDBOX.restore();
+      $$.SANDBOX.stub(MyDomainResolver.prototype, 'resolve').rejects();
+      const url = new SfdcUrl('https://login.salesforce.com');
+      try {
+        await shouldThrow(url.checkLightningDomain());
+      } catch (e) {
+        expect(e.name).to.equal('Error');
+      }
+    });
+  });
+
+  describe('Insecure HTTP warning', () => {
+    let emitWarningSpy: SinonSpy;
+    beforeEach(() => {
+      $$.SANDBOX.stub(MyDomainResolver.prototype, 'getCnames').resolves(TEST_CNAMES);
+      emitWarningSpy = spyMethod($$.SANDBOX, SfdcUrl.prototype, 'emitWarning');
+    });
+
+    afterEach(() => {
+      $$.SANDBOX.restore();
+      emitWarningSpy.restore();
```

Comment:
we shouldn't need either of the above calls since it's done for us by using `$$.SANDBOX`